main.go

```diff
@@ -151,34 +155,34 @@ func main() {
 		}
 	}()
 
+	var timer *time.Timer
+
 	for {
 		select {
-		case <-signalChan:
+		case <-termChan:
+			logger.Info("SIGTERM caught, gracefully stopping server...")
```

Comment:
Maybe debug level since most users won't appreciate what a SIGTERM is 😀

Reply:
Should I also change the logs in the `usr1Chan` case?

Reply:
I was actually pondering that, but decided those seem relatable since the user would have used that signal to trigger that part of the flow.